### PR TITLE
[AUDIO-HAL] [9.12.r1] [DONOTMERGE] cirrus_sony: Fast hack to use senary mi2s mixer

### PR DESCRIPTION
**Pull request by kholk**
On Sony EDO the cirrus codecs are put on the senary mi2s.
This commit is temporary and meant as a fast workaround: it will be
refactored properly in the future.

@MarijnS95 I have no time to do this properly. Do you?
If you don't, then we can just add the AUDIO_FEATURE_SONY_CIRRUS_SEN_MI2S
variable to device-sony-edo, merge this PR and then get this refactored in the near future.

### CODE REVIEW

File: hal/audio_extn/Android.mk [hunk 1]
```diff
@@ -325,6 +325,9 @@ LOCAL_CFLAGS += \
 
 ifeq ($(strip $(AUDIO_FEATURE_SONY_CIRRUS)),true)
 LOCAL_CFLAGS += -DTA_DEBUG -DDEBUG_SHOW_VALUES -DDEBUG_RUN_ONLY_READ_TA
+ifeq ($(strip $(AUDIO_FEATURE_SONY_CIRRUS_SEN_MI2S)),true)
+ LOCAL_CFLAGS += USE_SENARY_MI2S
```
Comment (MarijnS95) [inline hal/audio_extn/Android.mk:329]:
```suggestion
 LOCAL_CFLAGS += -DUSE_SENARY_MI2S
```

**Comment by MarijnS95** [comment]
Right, without this the logs clearly say that no backend is enabled:

```
02-10 10:19:33.970  8735  8752 E audio_hw_sony_cirrus_playback: cirrus_play_silence: Cannot start PCM_RX: cannot prepare channel: Invalid argument
02-10 10:19:33.970     0     0 E KONA Media1: ASoC: no backend DAIs enabled for KONA Media1
02-10 10:19:33.974     0     0 E q6asm_callback: cmd = 0x10bcd returned error = 0x1
02-10 10:19:33.974     0     0 E __q6asm_cmd: DSP returned error[ADSP_EFAILED] opcode 68557
```

Hardcoding and manually enabling/selecting the backend seems like a big hack anyway. Looking at `spkr_protection` it tries to be much smarter about actually selecting the right usecase and enabling its associated backend (as configured in audio_platform_info) and sound device - but we probably have to be careful not to enable too many mixer(s) (paths) that could conflict with what we set explicitly here. Perhaps we should use the same `USECASE_AUDIO_SPKR_CALIB_RX`, of which the `spkr-rx-calib` audiopath is empty on our side anyway (but we'll need to give it the appropriate backend).

Unfortunately both this patch and copypasting the "be smart about it" solution result in the same error:

```
02-10 10:20:52.589     0     0 I __afe_port_start: port id: 0x1018
02-10 10:20:52.589     0     0 E afe_get_cal_topology_id: cal_type 8 not initialized for this port 4120
02-10 10:20:52.589     0     0 E afe_get_cal_topology_id: cal_type 9 not initialized for this port 4120
02-10 10:20:52.589     0     0 I afe_send_port_topology_id: AFE set topology id 0x0  enable for port 0x1018 ret -22
02-10 10:20:52.589     0     0 I send_afe_cal_type: cal_index is 0
02-10 10:20:52.589     0     0 I send_afe_cal_type: dev_acdb_id[53] is 124
02-10 10:20:52.589     0     0 I afe_find_cal: cal_index 0 port_id 0x1018 port_index 53
02-10 10:20:52.589     0     0 I afe_find_cal: acdb_id 0 dev_acdb_id 124 sample_rate 0 afe_sample_rates 48000
02-10 10:20:52.589     0     0 I afe_find_cal: no matching cal_block found
02-10 10:20:52.589     0     0 E         : send_afe_cal_type cal_block not found!!
02-10 10:20:52.589     0     0 E afe_get_cal_hw_delay: Unable to find delay for sample rate 48000
02-10 10:20:52.589     0     0 I afe_send_hw_delay: port_id 0x1018 rate 48000 delay_usec 0 status -14
02-10 10:20:52.605  9125  9143 E audio_hw_sony_cirrus_playback: cirrus_play_silence: Cannot start PCM_RX: cannot start channel: Broken pipe
02-10 10:20:52.623     0     0 I afe_close: port_id = 0x1018
02-10 10:20:52.643  9125  9143 E audio_hw_sony_cirrus_playback: cirrus_stereo_calibration: SPK L Calibration failure, status: 0x0 0x0 0x0 0x2
02-10 10:20:52.643  9125  9143 E audio_hw_sony_cirrus_playback: cirrus_do_calibration: CRITICAL: Calibration failure
```

(Fwiw I checked with the "be smart about it" variant and it properly enables the backend DAI, `cirrus_get_mixer_value_by_name(CIRRUS_PLAYBACK_MIXER)` for `SEN` returns `1`!)

**EDIT**: A better middle-ground solution would be:

```c
audio_route_apply_and_update_path(adev->audio_route, "deep-buffer-playback");
```

And its `audio_route_reset_and_update_path` equivalent. That way you don't need to hardcode (with all the ifdefs and platform-specific toggles) what mixer to use to enable deep buffer playback. In addition it'd be nice to take over:

```c
int pcm_dev_rx_id = fp_platform_get_pcm_device_id(USECASE_AUDIO_PLAYBACK_DEEP_BUFFER, PCM_PLAYBACK);
```
To make sure that we select the right PCM ID now and in the future. It's likely always `0` though.

**Review by MarijnS95 — CHANGES_REQUESTED** [review]
Nice typo... But we are superseding this in a generic way anyway :)

**Comment by kholk** [comment]
Superseded by #20

**Pull request closed without merging**